app/controllers/products_controller.rb

```diff
@@ -0,0 +1,62 @@
+class ProductsController < ApplicationController
+  before_action :set_product, only: [:show, :edit, :update, :destroy]
+
+  def index
+    redirect_to root_path
+  end
+
+  # GET /products/new
+  def new
+    @product = Product.new
+  end
+
+  # GET /products/1
+  def show; end
+
+  # GET /users/1/edit
+  def edit; end
+
+  # POST /products
+  def create
+    @product = Product.new(product_params.except(:image))
+    @product.image.attach(product_params[:image])
+
+    respond_to do |format|
+      if @product.save
+        format.html { redirect_to root_path, notice: 'Product was successfully updated.' }
+      else
+        format.html { render :new }
+        format.json { render json: @product.errors, status: :unprocessable_entity }
```

Comment:
another way of writing an `if...else` block is something like: 

```
return format.html { redirect_to root_path, notice: 'Product was successfully added!' } if @product.save

format.html { render :new }
format.json { render json: @product.errors, status: :unprocessable_entity }
```
Also I think the message for `create` should be something like: 'Product was successfully added!' since it's created not updated.

Reply:
For some reason, the app gets stuck if i press the button to create a new item. It does the insertion in the database, but it doesn't call the redirect_to root_path. I have to manually click the "Back" button to see the results. Same for the update method.

Reply:
hmm I think in the index method you need to get all the products right? Something like:

`@products = Product.all`

I noticed this part is missing

Reply:
I have added that to the index method but it still doesn't work. I do get the error:
`No template found for ProductsController#update, rendering head :no_content`
And if I create a view called "update" it renders that one even though the route is set to the one in the if statement.

Reply:
It might be because you are doing `redirect_to root_path, notice: 'Product was successfully added!'`. You should do redirect_to @product, notice: 'Product was successfully added!' .